src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2227,6 +2232,8 @@ void MarkConstructorsUsedViaReflection (Collection<Instruction> instructions, st
 					Tracer.Push ($"Reflection-{method}");
 					try {
 						MarkMethod (method);
+						if (_context.AnnotateUnseenCallers)
```

Comment:
You need to add same to remaining UsedViaReflection* methods (e.g. for methods or accessors)